src/BayesFilters/src/AdditiveStateModel.cpp

```diff
@@ -0,0 +1,17 @@
+#include <BayesFilters/AdditiveStateModel.h>
+
+#include <Eigen/Dense>
+
+using namespace bfl;
+using namespace Eigen;
+
+void AdditiveStateModel::motion
+(
+    const Eigen::Ref<const Eigen::MatrixXf>& cur_states,
+    Eigen::Ref<Eigen::MatrixXf> mot_states
+)
+{
+    /* FIXME
+       Add additive noise generation. */
+    propagate(cur_states, mot_states); // + add noise
```

Comment:
Are we missing something here?

Reply:
I added a comment on the commit related to this file but probably isn't visible. A part from this, the code that is missing is already available in class `WhiteNoiseAcceleration`.

Since in this pull request `WhiteNoiseAcceleration` will inherits from `LinearStateModel` that in turn inherits from `AdditiveStateModel`, maybe we can move noise generation here once for all in order to have it for all the additive state models.

What do you think?

Reply:
> I added a comment on the commit related to this file because I wanted to discuss with you about it but probably isn't visible.

Unofrtunately, I don't see any comment 🤷‍♂ Maybe something wen wrong!





> maybe we can move noise generation here once for all in order to have it for all the additive state models.

Yes this is totally feasible. If we do like that, we must also add a virtual method to allow noise implementation. Am I right?

Reply:
The simplest way could be like

```
propagate(cur_states, mot_states);
mot_state += getSampleNoise();
```
where `getSampleNoise` is a pure virtual method in class `AdditiveStateModel`.

However, If we agree that the noise is always white and Gaussian we can also move the noise implementation inside `AdditiveStateModel` and let the user change it by overriding the non-pure virtual method `AdditiveStateModel::getSampleNoise`.

Maybe the first option is more clean. What do you think?

Reply:
I agree, it is better to have a simple implementation in "base" classes.